client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js

```diff
@@ -62,6 +68,79 @@ angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
         this.$scope.access.xdsl.description = data.description;
       }
     });
+
+    /*
+      HOW DIAGNOSTIC EVENTS WORK:
+
+      When we open the details modal, it sends a 'accessDiagnosticDetails:get'
+      event to retrieve a possibly loaded diagnostic.
+      We answer with 'accessDiagnosticDetails:arrived' and the diagnostic (or null)
+      as an event argument.
+      If the popup wants to relaunch a new one, it sends 'accessDiagnosticDetails:launch'.
+      We answer with 'accessDiagnosticDetails:arrived' when the diagnostic is finally there.
+
+      WHY WE DO THIS:
+
+      We use events because the process can be in any state when we arrive
+      on the page, and we don't know how long it will take for the diagnostic
+      to arrive (it is polled). If the user reloads the page, we lose controller's
+      state, so the code must be resilient and work even if we load the page in
+      the middle of the process.
+
+      Until the task is finished, the diagnostic we get from
+      the API is IRRELEVANT ans should not be displayed to the user.
+    */
+    this.$rootScope.$on('accessDiagnosticDetails:launch', () => {
+      this.launchDiagnostic();
+    });
+
+    this.$rootScope.$on('accessDiagnosticDetails:get', () => {
+      if (!this.accessDiagnostic && !this.$scope.loaders.accessDiagnosticLaunched) {
+        this.launchDiagnostic();
+      }
+      this.$rootScope.$broadcast('accessDiagnosticDetails:arrived', this.accessDiagnostic);
+    });
+
+    this.diagPollerTicket = this.XdslTaskPoller.register('accessDiagnosticRun', () => {
+      this.$scope.loaders.accessDiagnosticLaunched = false;
+      this.getDiagnostic();
+    });
+
+    this.additionalIpPollerTicket = this.XdslTaskPoller.register(
+      'pendingOrderAdditionalIpOption', () => {
+        this.getIps();
+        this.ordering = false;
+      },
+    );
+
+    this.$scope.$on('$destroy', () => {
+      this.XdslTaskPoller.unregister(this.diagPollerTicket);
+      this.XdslTaskPoller.unregister(this.additionalIpPollerTicket);
+    });
+
+    this.getDiagnostic();
+  }
+
+  openDetailsPopup() {
+    if (this.accessDiagnostic === null && !this.$scope.loaders.accessDiagnosticLaunched) {
```

Comment:
This test should be moved in `this.launchDiagnostic` method

Reply:
@euhmeuh plus the this.accessDiagnostic === null is useful ? You set `this.accessDiagnostic = null` in `launchDiagnostic` method.

Reply:
If it's null, launch it, otherwise just open the popup. It's a way to start it automatically when none has been started.